File: sample/src/main/java/vn/tiki/sample/base/BaseMvpActivity.java
package vn.tiki.sample.base;

import vn.tiki.architecture.mvp.Mvp;
import vn.tiki.architecture.mvp.MvpActivity;
import vn.tiki.daggers.ActivityInjector;

public abstract class BaseMvpActivity<V extends Mvp.View, P extends Mvp.Presenter<V>>
    extends MvpActivity<V, P> implements ActivityInjector {

  private ActivityDelegate activityDelegate = new ActivityDelegate();

  @Override protected void onPause() {
    super.onPause();
    activityDelegate.onPause(this);
  }

  @Override protected void onResume() {
    super.onResume();
    activityDelegate.onResume(this);
  }

  @Override public Object activityModule() {
    return activityDelegate.makeActivityModule(this);
  }
}

File: sample/src/main/java/vn/tiki/sample/base/BaseMvpActivity.kt
package vn.tiki.sample.base

import vn.tiki.architecture.mvp.Mvp
import vn.tiki.architecture.mvp.MvpActivity
import vn.tiki.daggers.ActivityInjector

abstract class BaseMvpActivity<V : Mvp.View, P : Mvp.Presenter<V>> : MvpActivity<V, P>(), ActivityInjector {

  private val activityDelegate = ActivityDelegate()

  override fun onPause() {
    super.onPause()
    activityDelegate.onPause(this)
  }

  override fun onResume() {
    super.onResume()
    activityDelegate.onResume(this)
  }

  override fun activityModule(): Any {
    return activityDelegate.makeActivityModule(this)
  }
}
